src/System.Diagnostics.EventLog/tests/EventLogTests/EventLogTests.cs

```diff
@@ -15,62 +16,314 @@ public void EventLogReinitializationException()
             {
                 eventLog.BeginInit();
                 Assert.Throws<InvalidOperationException>(() => eventLog.BeginInit());
+                eventLog.EndInit();
             }
         }
 
-        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotWindowsNanoServer))]
-        public void ClearLogTest()
+        [ConditionalFact(typeof(Helpers), nameof(Helpers.IsElevatedAndSupportsEventLogs))]
+        public void ClearLog()
         {
-            if (!AdminHelpers.IsProcessElevated())
-                return;
+            string log = "ClearTest";
+            string source = "Source_" + nameof(ClearLog);
 
-            string source = Guid.NewGuid().ToString("N");
-            string logName = Guid.NewGuid().ToString("N");
             try
             {
-                EventLog.CreateEventSource(source, logName);
-                using (EventLog myLog = new EventLog())
+                EventLog.CreateEventSource(source, log);
+                using (EventLog eventLog = new EventLog())
                 {
-                    myLog.Source = source;
-                    myLog.WriteEntry("Writing to event log.");
-                    Assert.True(myLog.Entries.Count != 0);
-                    myLog.Clear();
-                    Assert.Equal(0, myLog.Entries.Count);
+                    eventLog.Source = source;
+                    eventLog.Clear();
+                    Assert.Equal(0, eventLog.Entries.Count);
+                    eventLog.WriteEntry("Writing to event log.");
+                    Assert.Equal(eventLog.Entries.Count, 1);
```

Comment:
Expected result is always the first parameter, not hte second. Otherwise the messages are backwrads which is very confusing.